From: =?utf-8?q?Jonas_Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Sat, 28 Mar 2020 14:36:20 +0100
Subject: keyboard: Hide keyboardBox after destroying the keyboard

It seems there is a weird race condition between Clutter trying to
destroy the keyboard actor and Clutter trying to hide the keyboardBox
container actor: If the keyboardBox is hidden before destroying the
keyboard actor, Clutter doesn't repaint anything and the keyboard
remains visible until something else draws over it.

To fix this issue until we find the underlying Clutter bug, simply
destroy the keyboard actor before hiding the keyboardBox. The order in
which we call these doesn't matter anyway since hideKeyboard(true) hides
the keyboard immediately without an animation.

Fixes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/1736

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1142

Applied-Upstream: 3.36.1
Origin: https://gitlab.gnome.org/GNOME/gnome-shell/commit/867587ef
---
 js/ui/keyboard.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/keyboard.js b/js/ui/keyboard.js
index 61e36de..c70a791 100644
--- a/js/ui/keyboard.js
+++ b/js/ui/keyboard.js
@@ -1149,9 +1149,9 @@ var KeyboardManager = class KeyBoardManager {
             this._keyboard = new Keyboard();
         } else if (!enabled && this._keyboard) {
             this._keyboard.setCursorLocation(null);
-            Main.layoutManager.hideKeyboard(true);
             this._keyboard.destroy();
             this._keyboard = null;
+            Main.layoutManager.hideKeyboard(true);
         }
     }
 
